app/src/main/java/com/cornellappdev/android/eatery/ui/viewmodels/CompareMenusBotViewModel.kt

```diff
@@ -67,25 +76,13 @@ class CompareMenusBotViewModel @Inject constructor(
                         currentState.copy(
                             eateries =
                             (listOfNotNull(firstEatery) + eateriesApiResponse.data.filter { it.name != firstEatery?.name }).filter { eatery ->
-                                filters.all { filter ->
-                                    when (filter) {
-                                        is Filter.FromEatery -> filter.passesFilter(eatery)
-                                        is Filter.RequiresFavoriteEateries -> filter.passesFilter(
-                                            eatery,
-                                            favorites
-                                        )
-
-                                        is Filter.CustomFilter -> {
-                                            when (filter) {
-                                                Filter.CustomFilter.Selected -> selected.contains(
-                                                    eatery
-                                                )
-
-                                                Filter.CustomFilter.Today -> true
-                                            }
-                                        }
-                                    }
-                                }
+                                Filter.passesSelectedFilters(
+                                    compareMenusBottomSheetFilters,
+                                    filters,
+                                    FilterData(
+                                        eatery = eatery,
+                                        selectedEateryIds = selected.mapNotNull { it.id })
+                                )
```

Comment:
Ohhh I see so you basically pass the filter data in bulk to all the filters, that's pretty clean!! I like this structure a lot, reduces code reuse a ton overall!